### PR TITLE
[SDA-7474] Upgrading from pre release would fail to validate version

### PR DESCRIPTION
**Pull request by gdbranco**
Related issue: https://issues.redhat.com/browse/SDA-7474

# What
Upgrading from a channel group other than stable would fail to validate version

# Why
Channel group was not being taken into account for some validations

### CODE REVIEW

File: cmd/create/cluster/cmd.go [hunk 1]
```diff
@@ -2175,38 +2175,37 @@ func getAccountRolePrefix(roleARN string, role aws.AccountRole) (string, error)
...
-			}
+		if !valid {
+			return "", fmt.Errorf("version '%s' is not supported for hosted clusters", version)
 		}
-
-		version = "openshift-v" + version
 	}
 
-	return version, nil
+	return ocm.CreateVersionID(version, channelGroup), nil
```
Comment (gdbranco) [inline cmd/create/cluster/cmd.go:2208]: CreateVersionID does the prefix of "openshift-v" and adds the suffix for channel group if it is different than stable

File: cmd/upgrade/cluster/cmd.go [hunk 2]
```diff
@@ -201,7 +201,7 @@ func run(cmd *cobra.Command, _ []string) {
 	if isSTS {
 		r.Reporter.Infof("Ensuring account and operator role policies for cluster '%s'"+
 			" are compatible with upgrade.", cluster.ID())
-		err = roles.Cmd.RunE(roles.Cmd, []string{mode, cluster.ID(), version})
+		err = roles.Cmd.RunE(roles.Cmd, []string{mode, cluster.ID(), version, cluster.Version().ChannelGroup()})
```
Comment (gdbranco) [inline cmd/upgrade/cluster/cmd.go:204]: Now takes into consideration channel group

File: cmd/upgrade/roles/cmd.go [hunk 3]
```diff
@@ -109,9 +109,8 @@ func run(cmd *cobra.Command, argv []string) error {
 		aws.SetModeKey(argv[0])
 		ocm.SetClusterKey(argv[1])
 		skipInteractive = true
-		if len(argv) > 2 && argv[2] != "" {
-			args.clusterUpgradeVersion = argv[2]
-		}
+		args.clusterUpgradeVersion = argv[2]
```
Comment (gdbranco) [inline cmd/upgrade/roles/cmd.go:112]: Coming from cluster upgrade it should always have all the argvs, checking didn't make sense

File: pkg/ocm/versions.go [hunk 4]
```diff
@@ -248,9 +248,12 @@ func IsValidVersion(userRequestedVersion string, supportedVersion string, cluste
 	if err != nil {
 		return false, err
 	}
+
+	isPreRelease := a.Prerelease() != "" && b.Prerelease() != ""
+
 	versionSplit := a.Segments64()
 	//If user has specified patch we check directly and return the result
-	if len(versionSplit) > 2 && versionSplit[2] > 0 {
+	if len(versionSplit) > 2 && versionSplit[2] > 0 || isPreRelease {
```
Comment (gdbranco) [inline pkg/ocm/versions.go:256]: If the user has specified patch or is a pre release version we can compare directly

File: pkg/ocm/versions.go [hunk 5]
```diff
@@ -248,9 +248,13 @@ func IsValidVersion(userRequestedVersion string, supportedVersion string, cluste
...
 	}
+
+	isPreRelease := a.Prerelease() != "" && b.Prerelease() != ""
+
 	versionSplit := a.Segments64()
-	//If user has specified patch we check directly and return the result
-	if len(versionSplit) > 2 && versionSplit[2] > 0 {
+	// If user has specified patch or not specified patch but is a preRelease
+	// Check directly
+	if len(versionSplit) > 2 && versionSplit[2] > 0 || (versionSplit[2] == 0 && isPreRelease) {
```
Comment (gdbranco) [inline pkg/ocm/versions.go:257]: If it was a preRelease but patch was not 0 it was already taking into account. But if it is a pre release and patch is 0 we still need to check directly

**Review by hunterkepley — COMMENTED** [review]
/lgtm

**Pull request merged**